text_extensions_for_pandas/array/char_span.py

```diff
@@ -73,9 +73,19 @@ def __repr__(self) -> str:
                                        textwrap.shorten(self.covered_text, 80))
 
     def __eq__(self, other):
-        return (self.begin == other.begin
-                and self.end == other.end
-                and self.target_text == other.target_text)
+        if isinstance(other, CharSpan):
+            return (self.begin == other.begin
+                    and self.end == other.end
+                    and self.target_text == other.target_text)
+        elif isinstance(other, CharSpanArray):
+            return other == self
```

Comment:
isn't `self` an instance of `CharSpan`? So this is comparing a scalar to an array?

Reply:
I think comparison of an array and a scalar value is supposed to return a Boolean mask (that is, implicitly broadcast the scalar value to an array the same shape as the array argument) regardless of the order of operations. Are you aware of any documentation to the contrary?

Reply:
Oh ok, makes sense. It just looked a little strange at first glance